feature-libs/product-configurator/rulebased/components/attribute/types/single-selection-bundle-dropdown/configurator-attribute-single-selection-bundle-dropdown.component.ts

```diff
@@ -50,80 +34,26 @@ export class ConfiguratorAttributeSingleSelectionBundleDropdownComponent
     }
   }
 
-  get withQuantity() {
-    return this.quantityService.withQuantity(
-      this.attribute?.dataType ?? Configurator.DataType.NOT_IMPLEMENTED,
-      this.attribute?.uiType ?? Configurator.UiType.NOT_IMPLEMENTED
-    );
-  }
-
-  get disableQuantityActions() {
-    return this.quantityService.disableQuantityActions(
-      this.attributeDropDownForm?.value
-    );
-  }
-
-  onSelect(): void {
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: this.attributeDropDownForm?.value,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-    this.selectionChange.emit(event);
-  }
-
+  /**
+   TODO(issue: #11238): update @deprecated level to the release we are publishing with,
+   It is still 3.1 only because app.module.ts states that we are on 3.1.
+   Finally we must have 3.x, x>=2 here
+   */
+  /**
+   * @deprecated since 3.1
+   * User better onSelect('0')
+   */
   onDeselect(): void {
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: '0',
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-
-    this.selectionChange.emit(event);
+    this.onSelect('0');
```

Comment:
why this one set `'0'`, and others set `''`.

Reply:
`'0'` is sent to the engine in case of the drop-down list in order to set a value to `'No option selected'` 
`''` empty string is sent to the engine in order to deselect a value